zigzag/zigzag_test_log.py

```diff
@@ -222,10 +309,46 @@ def module_hierarchy(self):
         """
 
         if not self._module_hierarchy_facade:
-            self._module_hierarchy_facade = ModuleHierarchyFacade(self._classname, self._mediator)
+            self._module_hierarchy_facade = ModuleHierarchyFacade(self.classname, self._mediator)
 
         return self._module_hierarchy_facade.get_module_hierarchy()
 
+    @property
+    def test_file(self):
+        """Gets the test_file property
+
+        Returns:
+            str: the file containing the test that generated the xml
+        """
+        if self._test_file is None:
+            self._test_file = self._testcase_xml.attrib['file']
+
+        return self._test_file
+
+    @property
+    def classname(self):
+        """Gets the classname property
+
+        Returns:
+            str: the classname of the test
+        """
+        if self._classname is None:
+            self._classname = self._testcase_xml.attrib['classname']
+
+        return self._classname
+
+    @property
+    def def_line_number(self):
+        """Gets the def_line_number
+
+        Returns:
+            str: the sting that is the def_line_number
+        """
+        if self._def_line_number is None:
+            self._def_line_number = self._testcase_xml.attrib['line']
+
+        return self._def_line_number
+
     @property
     def qtest_test_log(self):
```

Comment:
Ooof. This is really long. Can we make some more properties to help break this up a bit? The rest of the class reads really nice up until this property.

Reply:
Yeah, I think I can do something about this.